### PR TITLE
feat: WeightedResultBar component

### PR DESCRIPTION
**Pull request by eszthoff**
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. Use imperative, present tense in your commit description ("change", not "changed" or "changes") without uppercases or period (.) at the end.

# Checklist
- [x] The implementation has been manually tested and complies with Textkernel [browser support guidelines](https://textkernel.com/browser-support/)
- [x] The implementation complies with [accessibility](CONTIRBUTING.md#accessibility) standards.
- [x] The component has a [displayName](CONTRIBUTING.md#display-names) defined.
- [x] The component comes with a detailed [PropTypes](CONTRIBUTING.md#component-props) (and defaultProps) definition.
- [x] Component PropTypes are sufficiently described / documented.
- [x] There is [a story](CONTRIBUTING.md#component-showcases) in Storybook.

### CODE REVIEW

File: src/components/WeightedResultBar/__tests__/WeightedResultBar.spec.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react';
+import toJson from 'enzyme-to-json';
+import { WeightedResultBar } from '../WeightedResultBar';
+
+describe('WeightedResultBar', () => {
+    it('should render correctly in loading state', () => {
+        const wrapper = mount(<WeightedResultBar>Result</WeightedResultBar>);
```
Comment (igorarkhipenko) [inline src/components/WeightedResultBar/__tests__/WeightedResultBar.spec.js:7]: The unit title is misleading here. Also there are no values passed to the component which breaks calculations in the snapshot below.

File: src/components/WeightedResultBar/__tests__/__snapshots__/WeightedResultBar.spec.js.snap [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      animated={false}
+      context="brand"
+      hidden={false}
+      small={true}
+    >
+      <div
+        aria-hidden={false}
+        aria-valuemax="100"
+        aria-valuemin="0"
+        aria-valuenow={NaN}
```
Comment (igorarkhipenko) [inline src/components/WeightedResultBar/__tests__/__snapshots__/WeightedResultBar.spec.js.snap:40]: ^ Right here

File: src/components/WeightedResultBar/__tests__/WeightedResultBar.spec.js (hunk 1 above)
Comment (eszthoff) [reply src/components/WeightedResultBar/__tests__/WeightedResultBar.spec.js:7]: uupps 👍

**Comment by carlobernardini** [comment]
I suppose transitions between loading to some value will not work this way. Can you confirm? How much would you need to change to make that work?

**Comment by eszthoff** [comment]
@carlobernardini this is a new requirement, can you please elaborate what you need. So far we never had transition between these states. If it is needed it can be done with some animation effect between mounting and unmounting as usual, like using react-transition-group

**Comment by carlobernardini** [comment]
See the demo in PR #428, (un)check `isLoading` in Storybook to see how it looks like.

**Comment by eszthoff** [comment]
@Carlo I added some code, so that the progress bar is now animated when either component is first rendered. So it will be there also if there is no loading state before the actual data is shown and vise versa. I hope that works for you.

**Pull request merged**